cloud/scope/vpc_cluster.go

```diff
@@ -486,9 +497,186 @@ func (s *VPCClusterScope) createVPC() (*vpcv1.VPC, error) {
 	} else if vpcDetails == nil {
 		return nil, fmt.Errorf("no vpc details after creation")
 	}
-	if err = s.TagResource(s.IBMVPCCluster.Name, *vpcDetails.CRN); err != nil {
+
+	// NOTE: This tagging is only attempted once. We may wish to refactor in case this single attempt fails.
+	if err = s.TagResource(s.Name(), *vpcDetails.CRN); err != nil {
 		return nil, fmt.Errorf("error tagging vpc: %w", err)
 	}
 
 	return vpcDetails, nil
 }
+
+// ReconcileVPCCustomImage reconciles the VPC Custom Image.
+func (s *VPCClusterScope) ReconcileVPCCustomImage() (bool, error) {
+	var imageID *string
+	// Attempt to collect VPC Custom Image info from Status.
+	if s.IBMVPCCluster.Status.Image != nil {
+		if s.IBMVPCCluster.Status.Image.ID != "" {
+			imageID = ptr.To(s.IBMVPCCluster.Status.Image.ID)
+		} else if s.IBMVPCCluster.Status.Image.Name != nil {
+			image, err := s.VPCClient.GetImageByName(*s.IBMVPCCluster.Status.Image.Name)
+			if err != nil {
+				return false, fmt.Errorf("error checking vpc custom image by name: %w", err)
+			}
+			// If the image was found via name, we should be able to get its ID.
+			if image != nil {
+				imageID = image.ID
+			}
+		}
+	} else if s.IBMVPCCluster.Spec.Image.CRN != nil {
+		// Parse the supplied Image CRN for Id, to perform image lookup.
+		imageCRN, err := crn.Parse(*s.IBMVPCCluster.Spec.Image.CRN)
+		if err != nil {
+			return false, fmt.Errorf("error parsing vpc custom image crn: %w", err)
+		}
+		if imageCRN.Resource == "" {
+			return false, fmt.Errorf("error parsing vpc custom image crn, missing resource id")
+		}
+		// If we didn't hit an error during parsing, and Resource was set, set that as the Image ID.
+		imageID = ptr.To(imageCRN.Resource)
+	}
+
+	// Check status of VPC Custom Image.
+	if imageID != nil {
+		image, _, err := s.VPCClient.GetImage(&vpcv1.GetImageOptions{
+			ID: imageID,
+		})
+		if err != nil {
+			return false, fmt.Errorf("error retrieving vpc custom image by id: %w", err)
+		}
+		if image == nil {
```

Comment:
Since you are passing id won't it return an error if the image is not found?
Please see whether we need this nil check?

Reply:
If the image is not found, I thought we got a 404 DetailedResponse back. I am defensively not expecting a non-error (`error == nil`) to result in `image` always being set.